website/docs/migration/from_state_notifier.mdx

```diff
@@ -0,0 +1,386 @@
+This paragraphs showcases a migration towards the new APIs going further than what's showcased
+in the [quickstart guide](/docs/from_provider/quickstart).
+
+All in all, migrating from `ChangeNotifier` to `AsyncNotifer` requires a paradigm shift, but it
+brings great simplification with the resulting migrated code.
+
+Take this example:
+```dart
+class MyChangeNotifier extends ChangeNotifier {
+  MyChangeNotifier() {
+    _init();
+  }
+  List<Todo> todos = [];
+  bool isLoading = true;
+  bool hasError = false;
+
+  Future<void> _init() async {
+    try {
+      // request mock
+      todos = Future.delayed(const Duration(seconds: 1), () => <Todo>[]);
+    } on Exception {
+      hasError = true;
+    } finally {
+      isLoading = false;
+      notifyListeners();
+    }
+  }
+
+  Future<void> addTodo(int id) async {
+    isLoading = true;
+    notifyListeners();
+
+    try {
+      // request mock
+      todos = Future.delayed(const Duration(seconds: 1), () => [Todo(id)]);
+      hasError = false;
+    } on Exception {
+      hasError = true;
+    } finally {
+      isLoading = false;
+      notifyListeners();
+    }
+  }
+}
+
+final myChangeProvider = ChangeNotifierProvider<MyChangeNotifier>((ref) {
+  return MyChangeNotifier();
+});
+```
+
+This implementation shows several faulty design choices such as:
+- The usage of `isLoading` and `hasError` to handle different asynchronous cases
+- The need to carefully handle requests with tedious `try`/`catch`/`finally` expressions
+- The need to inkove `notifyListeners` at the right times to make this implementation work
+- The presence of inconsistent or possibly undesirable states, e.g. initialization with an empty list
+
+While this example has been crafted to show how `ChangeNotifier` can lead to faulty design choices
+for newbie developers, the main takeaway is that mutable state might be harder than it looks.
+
+`Notifier`/`AsyncNotifer`, in combination with immutable state, can lead to better design choices
+and less errors; this approach lifts the developer from several intricacies due to the nature
+of asynchronous state.
+
+With `AsyncNotifier`, the above becomes:
+<AutoSnippet language="dart" {...fromChangeNotifier}></AutoSnippet>
+
+### Migration Process
+Let's analyze the migration process applied here:
+1. We've moved the initialization from a method invoked in a constructor, directly into `build`
+2. We've removed `todos`, `isLoading` and `hasError`: `state` suffice
+3. Notice how, since `build` returns a `Future<T>`, state is implicitly initialized with `AsyncLoading`,
+waiting for the initial request to finish
+4. We've returned the result of the (mocked) asynchronous request, with no try-catch-finally blocks
+5. We've then simplified `addTodo`, exploiting `AsyncValue`'s APIs
+6. Notice that, to apply mutations, we simply reassign `state` directly:
+listeners will be automatically notified about it
```

Comment:
It feels weird that in a document about migration, the migration bit is only 10 lines long.